src/extensions/shared/paste-spreadsheet-table.ts

```diff
@@ -0,0 +1,74 @@
+import { Extension } from '@tiptap/core'
+import { Plugin, PluginKey } from 'prosemirror-state'
+
+import { PASTE_EXTENSION_PRIORITY } from '../../constants/extension-priorities'
+import { parseHtmlToElement } from '../../helpers/dom'
+
+/**
+ * The `PasteSpreadsheetTable` extension adds the ability to paste a table copied from a spreadsheet
+ * web app (e.g., Google Sheets, Microsoft Excel), along with tables rendered by GitHub Flavored
+ * Markdown (GFM), into the editor.
+ *
+ * Since Typist does not yet support tables, this extension simply pastes the table as a string of
+ * paragraphs (one paragraph per row), with each cell separated by a space character. However,
+ * whenever we do add support for tables, this extension will need to be completely rewritten.
+ *
+ * Lastly, please note that formatting is lost when the copied table comes from Google Sheets or
+ * Microsoft Excel, because unfortunately, these apps style the cell contents using CSS.
+ */
+const PasteSpreadsheetTable = Extension.create({
+    name: 'pasteSpreadsheetTable',
+    priority: PASTE_EXTENSION_PRIORITY,
+    addProseMirrorPlugins() {
+        return [
+            new Plugin({
+                key: new PluginKey('pasteSpreadsheetTable'),
+                props: {
+                    transformPastedHTML(html) {
+                        // Attempt to extract a table HTML from the pasted HTML
+                        const tableHTML = /<table[^>]+>[\s\S]*?<\/table>/gi.exec(html)
+
+                        // Do not handle the event if a table was not found
+                        if (!tableHTML) {
+                            return html
+                        }
+
+                        const { firstElementChild: tableElement } = parseHtmlToElement(
+                            tableHTML[0],
```

Comment:
I tried copying two tables (from GitHub) and it only processed the first table. For example, this selection would only result in `C1 C2 ~~C3~~`:

![image](https://github.com/Doist/typist/assets/8531248/1d9fc5f9-d9b1-493a-a37d-e9499b8d2155)

Before this change, the output would be:

```
C1C2~~C3~~

H2H2H3_A1_A2A3
```

Should we support the case of multiple tables as well?

Reply:
It wasn't too hard or time consuming, so I added support for multiple (GFM) tables.